test/integration/xds_integration_test.cc

```diff
@@ -36,5 +40,49 @@ TEST_P(XdsIntegrationTest, RouterRequestAndResponseWithBodyNoBuffer) {
   testRouterRequestAndResponseWithBody(1024, 512, false);
 }
 
+typedef HttpProtocolIntegrationTest LdsIntegrationTest;
+
+INSTANTIATE_TEST_SUITE_P(Protocols, LdsIntegrationTest,
+                         testing::ValuesIn(HttpProtocolIntegrationTest::getProtocolTestParams(
+                             {Http::CodecClient::Type::HTTP1}, {FakeHttpConnection::Type::HTTP1})),
+                         HttpProtocolIntegrationTest::protocolTestParamsToString);
+
+// Sample test making sure our config framework correctly reloads listeners.
+TEST_P(LdsIntegrationTest, ReloadConfig) {
+  use_lds_ = true;
+  autonomous_upstream_ = true;
+  initialize();
+  fake_upstreams_[0]->set_allow_unexpected_disconnects(true);
+  const std::string lds_filename =
+      config_helper_.bootstrap().dynamic_resources().lds_config().path();
```

Comment:
Can we factor this kind of boiler plate outside of the individual tests. Ideally there are more declarative and have the file naming and renaming handled by some method which is just `applyUpdateConfig()` or something in the test fixture.